runtime/reference/ts_config_migration.md

```diff
@@ -0,0 +1,501 @@
+---
+title: "Configuring TypeScript"
+oldUrl:
+  - /runtime/manual/advanced/typescript/faqs/
+  - /runtime/manual/advanced/typescript/migration/
+  - /runtime/manual/advanced/typescript/configuration/
+  - /runtime/manual/advanced/typescript/types/
+  - /runtime/manual/typescript/types/
+  - /runtime/manual/advanced/faqs/
+---
+
+Deno’s flexibility shines in its equal treatment of TypeScript and JavaScript.
+Whether you’re transitioning from JavaScript to TypeScript or vice versa, Deno
+has features to ease the journey.
+
+## Type Checking JavaScript
+
+You may wish to make your JavaScript more type-sound without adding type
+annotations everywhere. Deno supports using the TypeScript type checker to type
+check JavaScript. You can mark individual files by adding the check JavaScript
+pragma to the file:
+
+```js
+// @ts-check
+```
+
+This will cause the type checker to infer type information about the JavaScript
+code and raise any issues as diagnostic issues.
+
+These can be turned on for all JavaScript files in a program by providing a
+configuration file with the check JS option enabled:
+
+```json
+{
+  "compilerOptions": {
+    "checkJs": true
+  }
+}
+```
+
+And then setting the `--config` option on the command line.
+
+## Using JSDoc in JavaScript
+
+When type-checking JavaScript or importing JavaScript into TypeScript, JSDoc
+annotations can provide additional type information beyond what can just be
+inferred from the code itself. Deno supports this seamlessly if you annotate
+your code inline with the supported
+[TypeScript JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).
+
+For example to set the type of an array use the following JSDoc comment:
+
+```js
+/** @type {string[]} */
+const a = [];
+```
+
+## Skipping type checking
+
+You might have TypeScript code that you are experimenting with, where the syntax
+is valid but not fully type safe. You can bypass type checking for a whole
+program by passing the `--no-check` flag.
+
+You can also skip whole files being type checked, including JavaScript if you
+have check JS enabled, by using the `nocheck` pragma:
+
+```js
+// @ts-nocheck
+```
+
+## Renaming JS files to TS files
+
+While this might work in some cases, it has some severe limits in Deno. This is
+because Deno, by default, runs type checking in what is called _strict mode_.
+This means a lot of unclear or ambiguous situations where are not caught in
+non-strict mode will result in diagnostics being generated, and JavaScript is
+nothing but unclear and ambiguous when it comes to types.
```

Comment:
A bit worried about the `it has some sever limits in Deno` phrasing. Makes it sound like it's something a user shouldn't do. We want to encourage users to go with TS. Maybe something like:

```suggestion
TypeScript files benefit from the TypeScript compiler being able to do more thorough safety checks of your code. This is often referred to as _strict mode_. When you rename a `.js` file to `.ts` you'll might see new type errors that you TypeScript wasn't able to detect before.
```
```